spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/secretmanager/SecretManagerCompatibilityTests.java

```diff
@@ -0,0 +1,85 @@
+package com.google.cloud.spring.autoconfigure.secretmanager;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretPayload;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import com.google.protobuf.ByteString;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.BootstrapRegistry.InstanceSupplier;
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ConfigurableApplicationContext;
+
+/**
+ * Unit tests to check compatibility of Secret Manager.
+ */
+class SecretManagerCompatibilityTests {
+
+  private static final String PROJECT_NAME = "hollow-light-of-the-sealed-land";
+
+  /**
+   * Users with the legacy configuration (i.e., bootstrap phrase) should get {@link
```

Comment:
It took me a while to figure out how this test works. Could you add a comment saying that configuration in `SecretManagerBootstrapConfigurationTests.TestBootstrapConfiguration` will be automatically picked up?

Also, I see now why testing scenario #3, in which neither old way nor new way exists is not currently possible in this repository -- the boostrap configuration will be loaded, no matter what, from spring.factories.

Reply:
done.